### PR TITLE
#165020080 enable user to login

### PR DESCRIPTION
**Pull request by dusmel**
#### What does this PR do?
This PR will enable users to login using an email and a password

#### Description of Task to be completed?
• gives the user an appropriate error message when credentials are not correct
• redirects the user to the home page on a successful login
• saves the user token in the local storage for other requests

#### How should this be manually tested?
using a browser go to `/login` add a valid `email` and `password` then submit

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#165020080](https://www.pivotaltracker.com/story/show/165020080)

#### Screenshots (if appropriate)

# 1
image

# 2
image

#### Questions:
N/A

### CODE REVIEW

File: src/__mocks__/axios.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// src/__mocks__/axios.ts
+
+// const mockAxios = jest.genMockFromModule('axios');
+
+// // this is the key to fix the axios.create() undefined error!
+// mockAxios.create = jest.fn(() => mockAxios);
+
+// export default mockAxios;
+// src/__mocks__/axios.ts
+
```
Comment (gracelungu) [inline src/__mocks__/axios.js]: @dusmel I think you should delete the commented code above.

File: src/api/wakanda.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import axios from 'axios';
+
+export default axios.create({
+  baseURL: process.env.REACT_APP_BACKEND_URL,
+  params: {
+    api_key: 'b81a990598f52450eb9f9ab81027dc56',
```
Comment (gracelungu) [inline src/api/wakanda.js]: @dusmel What is this API key?

**Review by gracelungu — CHANGES_REQUESTED** [review]
Good work @dusmel . Just implement the requested changes.

**Comment by Karlmusingo** [comment]
@dusmel, can please add a link to lead to the reset password page `/reset-password`
image

Also try to separate test cases with an empty line.

File: src/api/wakanda.js (hunk 2 above)
Comment (dusmel) [reply src/api/wakanda.js]: This was a placeholder for the token but since I am not using it directly let me remove it for now

File: src/__mocks__/axios.js (hunk 1 above)
Comment (dusmel) [reply src/__mocks__/axios.js]: coool 👍

**Review by gracelungu — APPROVED** [review]

**Review by coolbeatz71 — APPROVED** [review]
